### ISSUE TITLE
iOS 13 New UI close button doesn't reject promise

### ISSUE THREAD

**Issue by oleksandr-dziuban** [react-native-community/react-native-share#601]
### Steps to reproduce
On iOS 13 we have issue with new close button:
If user clicks on new iOS 13 UI close button - .catch() callback doesn't fire, so Promise is not rejected in this case. On iOS 12 with previous UI Cancel button it works correctly

### Expected behaviour
```catch``` callback should be called, promise should be rejected.

### Actual behaviour
```catch``` callback is not called, promise is not rejected.

### Environment
- **React Native version**: 0.61.2
- **React Native platform + platform version**: iOS 13.0

### react-native-share
**Version**: 2.0.0"

**Comment by YuraPostevka** [comment]
I have the same issue. Looks like IOS 13 updated Share menu has uncompleted functionality

**Comment by oleksandr-dziuban** [comment]
> I have the same issue. Looks like IOS 13 updated Share menu has uncompleted functionality

Yep, looks like for new iOS 13 UI "react-native-share" need a small patch to handle new "close" icon and throw an exception like on iOS 12

**Comment by MateusAndrade** [comment]
@oleksandr-dziuban I think we need a major release after fixing this. 🤔 

Also do you have any clue on the new "Modal" event can be handled?

**Comment by oleksandr-dziuban** [comment]
@MateusAndrade Yes, I think new major version is OK.
New Modal works fine, just we have this issue with close button + swipe down gesture.
Also one more weird issue: when we select "Save Image" we can see rejected Promise and error
```"No callback found with cbID 6377 and callID 3188 for ActionsSheetManager.showShareActionSheeyWithOptions - most likely callback was already invoked"```
I think this is broken behavior too now with iOS 13

**Comment by oleksandr-dziuban** [comment]
![2019-10-07_1922](https://user-images.githubusercontent.com/5430640/66329882-f8e8f300-e937-11e9-96b3-83b566334dfd.png)

**Comment by oleksandr-dziuban** [comment]
New Share Menu is one of the biggest breaking changes in iOS 13

**Comment by zachrnolan** [comment]
Also experiencing this, any ideas on a fix? We aren't able to implement this library on our app because of the crash.

**Comment by oleksandr-dziuban** [comment]
Just library should be fixed according to latest iOS 13 changes. For this we need ask maintainers to fix this

**Comment by jgcmarins** [comment]
Anyone interested on working on this fix?

**Comment by toddychen** [comment]
Same issue here. The broken behavior for `Save Image` and `Save to Files` options. iOS 13.

**Comment by tomtargosz** [comment]
@jgcmarins I'll open a PR for the promise not resolving when the sharesheet is dismissed by the user. After that a new release needs to be made, #584 and #586 both fixed crashes on iOS 13 but haven't made it into a new release.

**Comment by oleksandr-dziuban** [comment]
@tomtargosz Thanks a lot!

**Comment by oleksandr-dziuban** [comment]
@tomtargosz Could you please publish the new version?

**Comment by jgcmarins** [comment]
We plan to release a new version as soon as possible.
Could you use master branch to check if everything is fine?

**Comment by Simek** [comment]
@tomtargosz When the new version with this fix will be available?

**Comment by tomtargosz** [comment]
@Simek I'm just a contributor so I don't have the permissions to create a release. @oleksandr-dziuban can you confirm that your issues are now resolved with the fixes merged into master? After that @jgcmarins should be good to make a new release 👍

**Comment by Simek** [comment]
@tomtargosz I can confirm that code from `master` do not produce error in our project based on RN 60.6 on iOS 13.1.3.

**Comment by oleksandr-dziuban** [comment]
@tomtargosz Same for me, issue fixed, please publish new release. Thanks

**Comment by MateusAndrade** [comment]
@oleksandr-dziuban, @Simek, @tomtargosz  thanks for letting us know that the master version looks stable. I got really busy on the last weeks, and cannot give the proper attention to ```react-native-share```. 

If anybody can help me testing the master version, i can plan a release on the next week. Please, let me know if anynone find anything related to the master version that can be a breaking change or even a bug. :smiley:

**Comment by esutton** [comment]
Easy to reproduce on iOS 13:

1. On share dialolg, select **Messages** like you are about to SMS text PDF
2. Press **Cancel** (or **Send**)

````
2019-11-06 12:37:09.327 [fatal][tid:main] Illegal callback invocation from native module. This callback type only permits a single invocation from native code.
2019-11-06 12:37:09.327451-0600 UtiliGuard[14029:3724994] Illegal callback invocation from native module. This callback type only permits a single invocation from native code.
````

Using:
````
    "react": "^16.9.0",
    "react-native": "0.60.6",
    "react-native-pdf": "5.1.7",
````

**Comment by Simek** [comment]
@esutton Unfortunately since that fix has been merged there was no release published. 

Until then you can replace `react-native-share` version string with `http://react-native-community/react-native-share` (this will use `master` branch from GitHub repository which include the fix but may be unstable in long run).

More information: https://docs.npmjs.com/files/package.json#git-urls-as-dependencies

**Comment by esutton** [comment]
@Simek Thanks. 

Stability was definitely a concern so I used the commit hash in my package.json

Master fixed it at d88e542ddd0983d09a4aa1a82737bb05b5731801
````
    "react-native-share": "https://github.com/react-native-community/react-native-share#d88e542ddd0983d09a4aa1a82737bb05b5731801",
````

**Comment by Simek** [comment]
@esutton No problem. Just a tip - you can simplify your version string to:
```
    "react-native-share": "react-native-community/react-native-share#d88e542",
```
